### PR TITLE
Enable equiangular maps in BBH domain

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Another step towards #3545.

### Upgrade instructions

In input files using the `BinaryCompactObject` domain creator, add the `UseEquiangularMap: True` option.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Domain/DomainHelpers.cpp [hunk 1]
```diff
@@ -717,7 +717,8 @@ std::vector<domain::CoordinateMaps::Frustum> frustum_coordinate_maps(
         use_equiangular_map,
         projective_scale_factor,
         false,
-        sphericity});
+        sphericity,
+        -1.});
```
Comment (mar-celine) [inline src/Domain/DomainHelpers.cpp]: Can have be `-1.0` and `1.0` on line 736 for consistency with the rest of the file. Also for clarity for the last argument (since the other arguments have variable names) can add a comment on line 702 (The frustums on the left are given a `-1.0` indicating a lower half equiangular map while the frustums on the right are given a `1.0` indicating an upper half equiangular map) (Ideally this should probably be an enum, but that was my own mistake!)

File: src/Domain/Creators/BinaryCompactObject.hpp [hunk 2]
```diff
@@ -589,7 +596,8 @@ class BinaryCompactObject : public DomainCreator<3> {
       double outer_radius_domain,
       const typename InitialRefinement::type& initial_refinement,
       const typename InitialGridPoints::type& initial_number_of_grid_points,
-      bool use_projective_map = true, double frustum_sphericity = 0.0,
+      bool use_equiangular_map = true, bool use_projective_map = true,
+      double frustum_sphericity = 0.0,
```
Comment (mar-celine) [inline src/Domain/Creators/BinaryCompactObject.hpp]: Minor comment here and for line 626: Now that we have equiangular capability, and the frustum layer can be made spherical, should this default value for the `frustum_sphericity` be kept at `0.0`?  Maybe it can be changed to be `1.0` now unless there is a good reason to avoid using that value by default.

File: src/Utilities/CartesianProduct.hpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
```
Comment (mar-celine) [inline src/Utilities/CartesianProduct.hpp:1]: This would be good as a separate commit because it is conceptually unrelated to the domain changes.

**Review by mar-celine — COMMENTED** [review]
Looks good! Just some minor comments

File: src/Utilities/CartesianProduct.hpp (hunk 3 above)
Comment (nilsvu) [reply src/Utilities/CartesianProduct.hpp:1]: It is a separate commit :)

**Review by mar-celine — DISMISSED** [review]
Looks good to me!

File: src/Utilities/CartesianProduct.hpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    // bind the first free variable in the lambda, and
+    // keep one free variable for each of the remaining ranges
+    detail::insert_tuples(
+        [&op, &it](InputIterator2... elems) { op(it, elems...); }, tail...);
+  }
+}
+
+}  // namespace detail
+
+// convert a tuple of ranges to the range of tuples representing the Cartesian
```
Comment (nilsdeppe) [inline src/Utilities/CartesianProduct.hpp]: Why no Doxygen comment here?

File: src/Utilities/CartesianProduct.hpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+namespace detail {
+
+// the lambda is fully bound with one element from each of the ranges
+template <class Op>
+void insert_tuples(Op op) {
+  // evaluating the lambda will insert the currently bound tuple
+  op();
+}
+
+// "peal off" the first range from the remaining tuple of ranges
```
Comment (nilsdeppe) [inline src/Utilities/CartesianProduct.hpp]: `peal` -> `peel`, same below

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash and rebase immediately

**Review by nilsdeppe — APPROVED** [review]
